code/src/test/kotlin/com/expediagroup/sdk/core/http/ResponseTest.kt

```diff
@@ -0,0 +1,343 @@
+            .protocol(Protocol.HTTP_1_1)
+            .addHeader("Header1", "Value1")
+            .addHeader("Header2", "Value2")
+            .removeHeader("Header1")
+            .build()
+
+        // When &  Expect
+        assertNull(response.headers.get("Header1"))
+        assertEquals("Value2", response.headers.get("Header2"))
+    }
+
+    @Test
+    fun `should indicate if response is successful`() {
+        // Given
+        val responseStatus200 = Response
+            .builder()
+            .request(Request.Builder().method("GET").url("https://example.com").build())
+            .protocol(Protocol.HTTP_1_1)
+            .status(Status.OK)
+            .message("OK")
+            .build()
+
+        val responseStatus100 = Response
+            .builder()
+            .request(Request.Builder().method("GET").url("https://example.com").build())
+            .protocol(Protocol.HTTP_1_1)
+            .status(Status.CONTINUE)
+            .message("OK")
+            .build()
+
+        val responseStatus400 = Response
+            .builder()
+            .request(Request.Builder().method("GET").url("https://example.com").build())
+            .protocol(Protocol.HTTP_1_1)
+            .status(Status.NOT_FOUND)
+            .message("Not Found")
+            .build()
+
+        // When & Expect
+        assertTrue(responseStatus200.isSuccessful)
+        assertFalse(responseStatus400.isSuccessful)
+        assertFalse(responseStatus100.isSuccessful)
+    }
+
+    @Test
+    fun `should throw exception when required fields are missing`() {
+        // Given
+        val builder = Response.builder()
+
+        // When & Expect
+        assertThrows<IllegalStateException> { builder.build() }.also {
+            assertEquals("request is required", it.message)
+        }
+
+        builder.request(Request.Builder().method("GET").url("https://example.com").build())
+        assertThrows<IllegalStateException> { builder.build() }.also {
+            assertEquals("protocol is required", it.message)
+        }
+
+        builder.protocol(Protocol.HTTP_1_1)
+        assertThrows<IllegalStateException> { builder.build() }.also {
+            assertEquals("status is required", it.message)
+        }
+
+        builder.status(Status.OK)
+        assertThrows<IllegalStateException> { builder.build() }.also {
+            assertEquals("message is required", it.message)
+        }
+    }
+
+    @Test
+    fun `should close response body`() {
+        // Given
+        val body = ResponseBody.create("Test body".byteInputStream())
+        val response = Response.Builder()
+            .request(Request.Builder().method("GET").url("https://example.com").build())
+            .protocol(Protocol.HTTP_1_1)
+            .status(Status.OK)
+            .message("OK")
+            .body(body)
+            .build()
+
+        // When
+        response.close()
+
+        // Expect
+        assertFalse(response.body?.source()?.isOpen == true)
```

Comment:
Wouldn't just work if we replaced the condition with?...
```kotlin
assertFalse(response.body?.source()?.isOpen)
```

Reply:
No, Kotlin treats `Boolean` & `Boolean?` differently